sdk/keyvault/keyvault-admin/review/keyvault-admin.api.md

```diff
@@ -0,0 +1,98 @@
+## API Report File for "@azure/keyvault-admin"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { PagedAsyncIterableIterator } from '@azure/core-paging';
+import { TokenCredential } from '@azure/core-http';
+
+// @public (undocumented)
+export class AccessControlClient {
+    constructor(vaultUrl: string, credential: TokenCredential, pipelineOptions?: AccessControlClientOptions);
+    createRoleAssignment(scope: RoleAssignmentScope, name: string, options?: CreateRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    deleteRoleAssignment(scope: RoleAssignmentScope, name: string, options?: DeleteRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    getRoleAssignment(scope: RoleAssignmentScope, name: string, options?: DeleteRoleAssignmentOptions): Promise<KeyVaultRoleAssignment>;
+    listRoleAssignments(scope: RoleAssignmentScope, options?: ListRoleAssignmentsOptions): PagedAsyncIterableIterator<KeyVaultRoleAssignment>;
+    listRoleDefinitions(scope: RoleAssignmentScope, options?: ListRoleDefinitionsOptions): PagedAsyncIterableIterator<KeyVaultRoleDefinition>;
+    readonly vaultUrl: string;
+}
+
+// @public
+export interface AccessControlClientOptions extends coreHttp.PipelineOptions {
+    serviceVersion?: SUPPORTED_API_VERSIONS;
+}
+
+// @public
+export interface CreateRoleAssignmentOptions extends coreHttp.OperationOptions {
+    properties?: RoleAssignmentProperties;
+}
+
+// @public
+export interface DeleteRoleAssignmentOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface KeyVaultRoleAssignment {
+    readonly id?: string;
+    readonly name?: string;
+    properties?: RoleAssignmentPropertiesWithScope;
+    readonly type?: string;
+}
+
+// @public
+export interface KeyVaultRoleDefinition {
+    assignableScopes?: string[];
+    description?: string;
+    readonly id?: string;
+    readonly name?: string;
+    permissions?: RoleDefinitionPermission[];
+    roleName?: string;
+    roleType?: string;
+    readonly type?: string;
+}
+
+// @public
+export const LATEST_API_VERSION = "7.1";
+
+// @public
+export interface ListRoleAssignmentsOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface ListRoleDefinitionsOptions extends coreHttp.OperationOptions {
+}
+
+// @public
+export interface RoleAssignmentProperties {
+    principalId?: string;
+    roleDefinitionId?: string;
+}
+
+// @public
+export interface RoleAssignmentPropertiesWithScope extends RoleAssignmentProperties {
+    scope?: RoleAssignmentScope;
+}
+
+// @public
+export type RoleAssignmentScope = "/" | "/keys";
```

Comment:
`RoleAssignmentScope` can be a specific resource Id as well. [example](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/keyvault/Azure.Security.KeyVault.Administration/samples/Sample2_RbacScopeAssignment.md#assigning-a-role-to-a-specific-key-scope)

Reply:
This is great feedback! Thank you!

Reply:
Would this be ok? `"/" | "/keys" | string` ?

Reply:
That would work, as long as we explain usage in the api comments and samples. We created a custom type in .NET just to make this a bit more clear.

Reply:
Well, I meant a custom type like this:

```ts
/**
 * A scope of the role assignment.
 * The valid scopes are "/", "/keys" any a specific resource Id.
 */
export type RoleAssignmentScope = "/" | "/keys" | string;
```

How does  that look? 🌞

Reply:
nit: `any specific Key resource Id/`

Reply:
Made the comment this: `The valid scopes are: "/", "/keys" and any a specific resource Id followed by a slash, as in "ID/".` 🌞 more feedback always appreciated!